shared/src/main/scala/fr/hmil/scalahttp/Method.scala

```diff
@@ -17,14 +13,42 @@ final case class Method private (name: String) {
   override def hashCode: Int = name.toUpperCase.hashCode
 }
 
+/** Exposes available methods as object as well as an implicit conversion
+  * from string to Method objects.
+  *
+  * Because all backends do not support all methods, this library imposes a subset
+  * of all available HTTP Methods. Should you find a use case for this library
+  * with other HTTP methods, please submit an issue with your motivation.
+  */
 object Method {
-  val OPTIONS = Method("OPTIONS")
+
   val GET = Method("GET")
-  val HEAD = Method("HEAD")
   val POST = Method("POST")
+  val HEAD = Method("HEAD")
+  val OPTIONS = Method("OPTIONS")
   val PUT = Method("PUT")
   val DELETE = Method("DELETE")
   val TRACE = Method("TRACE")
-  val CONNECT = Method("CONNECT")
-  implicit def fromString(name: String): Method = new Method(name)
+
+  /** Transform a method string into a [[Method]] instance.
+    *
+    * Given that this library is very unlikely to be
+    * used with other more exotic methods, if the user specifies an unsupported method,
+    * it is probably a typo rather than an actual HTTP method. We therefore report the
+    * error early as an IllegalArgumentException
+    *
+    * @param name "GET", "POST", "HEAD", "OPTIONS", "PUT", "DELETE" or "TRACE"; Case insensitive.
+    * @throws IllegalArgumentException when name is none of the legal values.
+    * @return The method instance for the provided method name.
+    */
+  implicit def fromString(name: String): Method = name.toUpperCase match {
```

Comment:
I think accepting strings is just being unsafe for the sake of it.

If you're concerned about a transition path, you could provide this implicit conversion in a separate `Implicits` object, so that users can `import Method.Implicits._` if they want to use strings.